Wikipedia/Code/SectionEditorViewController.swift

```diff
@@ -346,31 +341,26 @@ extension SectionEditorViewController: UIScrollViewDelegate {
 
 extension SectionEditorViewController: SectionEditorNavigationItemControllerDelegate {
     func sectionEditorNavigationItemController(_ sectionEditorNavigationItemController: SectionEditorNavigationItemController, didTapProgressButton progressButton: UIBarButtonItem) {
-        if changesMade {
-            messagingController.getWikitext { [weak self] (result, error) in
-                
-                guard let self = self else { return }
-                
-                if let error = error {
-                    assertionFailure(error.localizedDescription)
+        messagingController.getWikitext { [weak self] (result, error) in
+
+            guard let self = self else { return }
+
+            if let error = error {
+                assertionFailure(error.localizedDescription)
+                return
+            } else if let wikitext = result {
+                guard let vc = EditPreviewViewController.wmf_initialViewControllerFromClassStoryboard() else {
                     return
-                } else if let wikitext = result {
-                    guard let vc = EditPreviewViewController.wmf_initialViewControllerFromClassStoryboard() else {
-                        return
-                    }
-                    self.inputViewsController.resetFormattingAndStyleSubmenus()
-                    self.needsSelectLastSelection = true
-                    vc.theme = self.theme
-                    vc.section = self.section
-                    vc.wikitext = wikitext
-                    vc.delegate = self
-                    vc.funnel = self.editFunnel
-                    self.navigationController?.pushViewController(vc, animated: true)
                 }
+                self.inputViewsController.resetFormattingAndStyleSubmenus()
+                self.needsSelectLastSelection = true
+                vc.theme = self.theme
+                vc.section = self.section
+                vc.wikitext = wikitext
+                vc.delegate = self
+                vc.funnel = self.editFunnel
+                self.navigationController?.pushViewController(vc, animated: true)
             }
-        } else {
-            let message = WMFLocalizedString("wikitext-preview-changes-none", value: "No changes were made to be previewed.", comment: "Alert text shown if no changes were made to be previewed.")
```

Comment:
Seems like this should stay and at least do a string comparison to the original wikitext to verify that something was changed - currently it allows the user to go through the process and fails silently

Reply:
the `progressButton` should be disabled if no changes were made - what steps do you take to go through and fail silently?